portfolio/src/main/webapp/script.js

```diff
@@ -33,76 +33,120 @@ function addRandomFact() {
+  }, {
+    img: 'mountain.jpg',
+    headingColor: 'midnightblue',
+    headingBgColor: 'rgba(0,128,128,.15)',
+  }, {
+    img: 'balloons.jpg',
+    headingColor: 'indianred',
+    headingBgColor: 'rgba(255,215,0,.2)',
+  }, {
+    img: 'iceberg.jpg',
+    headingColor: 'darkslategray',
+    headingBgColor: 'rgba(72,61,139,.2)',
+  }, {
+    img: 'grass.jpg',
+    headingColor: 'green',
+    headingBgColor: 'rgba(173,255,47,.2)',
+  },
+];
 
 /**
- * Advances the background image by a given offset from the current image 
+ * Advances the background image by a given offset from the current image.
  */
 function changeBackground(offset) {
-  // List of background images and their color themes
-  const backgroundImgs = [
-    {
-      img: 'forest.jpg',
-      headingColor: '#523029',
-      headingBgColor: 'rgba(160,82,45,.1)',
-    }, {
-      img: 'ocean.jpg',
-      headingColor: 'navy',
-      headingBgColor: 'rgba(65,105,225,.1)',
-    }, {
-      img: 'mountain.jpg',
-      headingColor: 'midnightblue',
-      headingBgColor: 'rgba(0,128,128,.15)',
-    }, {
-      img: 'balloons.jpg',
-      headingColor: 'indianred',
-      headingBgColor: 'rgba(255,215,0,.2)',
-    }, {
-      img: 'iceberg.jpg',
-      headingColor: 'darkslategray',
-      headingBgColor: 'rgba(72,61,139,.2)',
-    }, {
-      img: 'grass.jpg',
-      headingColor: 'green',
-      headingBgColor: 'rgba(173,255,47,.2)',
-    },
-  ];
-
-  // Update the index of the current background image
-  curBackground = (curBackground + offset) % backgroundImgs.length;
+  // Update the index of the current background image.
+  currentBackgroundIndex = (currentBackgroundIndex + offset) % backgroundImgs.length;
 
-  // Wrap around to the last image if the index is negative
-  if (curBackground < 0) {
-    curBackground += backgroundImgs.length;
+  // Wrap around to the last image if the index is negative.
+  if (currentBackgroundIndex < 0) {
+    currentBackgroundIndex += backgroundImgs.length;
   }
 
-  const newBackground = backgroundImgs[curBackground];
+  const newBackground = backgroundImgs[currentBackgroundIndex];
 
-  // Update the page with the new background image
+  // Update the page with the new background image.
   const html = document.documentElement;
   html.style.backgroundImage = `url(images/${newBackground.img})`;
 
-  // Update the headings to the color theme for the new image
+  // Update the headings to the color theme for the new image.
   updateColorTheme(newBackground.headingColor, newBackground.headingBgColor);
 }
 
 /**
- * Updates the heading color CSS properties
+ * Updates the heading color CSS properties.
  */
 function updateColorTheme(headingColor, headingBgColor) {
   document.body.style.setProperty('--heading-color', headingColor);
   document.body.style.setProperty('--heading-bg-color', headingBgColor);
+}
+
+/**
+ * Fetches the list of comments from the server and adds them to the DOM.
+ */
+function getComments() {
```

Comment:
You may want to use 'async' and 'await' so that you can eliminate using 'then' and the code is a little easier to read.